src/main/java/io/fabric8/elasticsearch/cloud/kubernetes/KubernetesAPIServiceImpl.java

```diff
@@ -28,20 +32,29 @@
 
   private final String namespace;
   private final String serviceName;
+  private final String podLabel;
 
   @Override
   public Endpoints endpoints() {
     logger.debug("get endpoints for service {}, namespace {}", serviceName, namespace);
     return client().endpoints().inNamespace(namespace).withName(serviceName).get();
   }
 
+  @Override
+  public List<Pod> pods() {
+    logger.debug("get endpoints with pod label {}, namespace {}", podLabel, namespace);
+    final String[] l = podLabel.split("=");
+    return client().pods().inNamespace(namespace).withLabel(l[0], l[1]).list().getItems();
```

Comment:
As long as we control how we get the key/value and know that's what it will be, this is sufficient for me.  We can provide further updates if we think we will need them.